### ISSUE TITLE
Domains set to be incorrectly recreated to add "type=kvm" property

### ISSUE THREAD

**Issue by rombert** [dmacvicar/terraform-provider-libvirt#1062]
### Linux distribution

``` openSUSE Tumbleweed ```

### Terraform version

```sh
Terraform v1.5.5
```

### Provider and libvirt versions

```sh
v0.7.6
```
___

## Checklist

- [ ] Is your issue/contribution related with enabling some setting/option exposed by libvirt that the plugin does not yet support, or requires changing/extending the provider terraform schema?

  - [ ] Make sure you explain why this option is important to you, why it should be important to everyone. Describe your use-case with detail and provide examples where possible.
  - [ ] If it is a very special case, consider using the _XSLT_ support in the provider to tweak the definition instead of opening an issue
  - [ ] Maintainers do not have expertise in every libvirt setting, so please, describe the feature and how it is used. Link to the appropriate documentation

- [x] Is it a bug or something that does not work as expected? Please make sure you fill the version information below:

### Setup

I have a very plain terraform setup, coming from the 0.6.x days. The domain definition looks like this

```
# domain (VM definition)
resource "libvirt_domain" "microos_domain" {
  for_each  = { for vm_def in var.vm_defs : vm_def.hostname => vm_def }
  name      = "homelab002-${each.key}"
  autostart = true

  cpu {
    mode = "host-passthrough"
  }

  memory = each.value.memory
  vcpu   = each.value.cpu_count

  disk {
    volume_id = libvirt_volume.os_volume[each.key].id
  }

  network_interface {
    network_name   = libvirt_network.homelab002_net.name
    hostname       = data.template_file.ignition_data[each.key].vars.hostname
    wait_for_lease = true
  }
```

I upgraded from 0.6.14 to 0.7.6 and `terraform plan` now wants to recreate my domains, e.g.

```
  # libvirt_domain.microos_domain["worker-1"] must be replaced
-/+ resource "libvirt_domain" "microos_domain" {
      ~ arch            = "x86_64" -> (known after apply)
      - cmdline         = [] -> null
      ~ emulator        = "/usr/bin/qemu-system-x86_64" -> (known after apply)
      ~ id              = "5c6ce085-2f92-4af9-966e-a04a3aaa1567" -> (known after apply)
      ~ machine         = "pc-i440fx-6.2" -> (known after apply)
        name            = "homelab002-worker-1"
      + type            = "kvm" # forces replacement
        # (7 unchanged attributes hidden)

      ~ network_interface {
          ~ addresses      = [
              - "10.25.1.107",
            ] -> (known after apply)
          ~ mac            = "52:54:00:E6:3A:DC" -> (known after apply)
          ~ network_id     = "a7677a48-27ad-4769-8af2-504d9cb396ef" -> (known after apply)
            # (3 unchanged attributes hidden)
        }

        # (2 unchanged blocks hidden)
    }
```

I checked the VM definitions via virt-manager and they all start with

```xml
<domain xmlns:qemu="http://libvirt.org/schemas/domain/qemu/1.0" type="kvm">
  <name>homelab002-worker-1</name>
```

### Steps to Reproduce Issue

I guess start with 0.6.x and upgrade to 0.7.x?

## Additional information:

Do you have SELinux or Apparmor/Firewall enabled? Some special configuration?
No.
Have you tried to reproduce the issue without them enabled?
N/A

This behaviour seems to have been brought in via #1045

**Comment by michaelbeaumont** [comment]
Yeah, it was probably a mistake to set the default to be `kvm`. Can you set it to the empty string to avoid the recreation?

**Comment by rombert** [comment]
> Yeah, it was probably a mistake to set the default to be `kvm`. Can you set it to the empty string to avoid the recreation?

Thanks for the hint. With that property set the domains are no longer set to be re-created.

My immediate problem is solved, I'll leave this up to you whether you want to follow-up with a fix or just close.

**Comment by scabala** [comment]
Since the PR for providing a property is merged, I think this can be closed?

**Comment by rombert** [comment]
@scabala - do you mean https://github.com/dmacvicar/terraform-provider-libvirt/pull/1045 ? That was the one that caused the problem for me. I have a workaround but the root cause is not fixed.

**Comment by scabala** [comment]
Oh, I misunderstood the issue.

I think this is part of bigger idea of having non-recreating updates to domains like #1069

**Comment by dmacvicar** [comment]
ℹ️ ℹ️ ℹ️ ℹ️ ℹ️ ℹ️

This issue report is relevant to the legacy version of the provider, which is now in the v0.8 branch of this repository.

Future development is based on a new provider, which is not compatible with this one, nor does share code.

As the new provider solves many issues with the legacy, and to make development in my free time more enjoyable, I have decided to close all bugs for the legacy provider, and to ask to check if the bug would apply to the new one. This also to encourage trying the new version.

- https://registry.terraform.io/providers/dmacvicar/libvirt/latest

and check the documentation:

- https://github.com/dmacvicar/terraform-provider-libvirt/blob/main/README.md
- https://registry.terraform.io/providers/dmacvicar/libvirt/latest/docs

You are free to reopen the bug for v0.8. It may encourage someone to fix it. My efforts will go into the new provider.

I ask you to check the new provider and re-evaluate this bug. 🙏
